File: esim-frontend/lib/features/offers/models/offer.dart
class Offer {
  const Offer({
    required this.id,
    required this.country,
    required this.region,
    required this.destination,
    required this.category,
    required this.title,
    required this.description,
    required this.popularity,
    required this.dataVolume,
    required this.validityDays,
    required this.price,
    required this.createdAt,
    required this.updatedAt,
  });

  final int id;
  final String country;
  final String region;
  final String destination;
  final String category;
  final String title;
  final String description;
  final String popularity; // 'HIGH' | 'MEDIUM' | 'LOW'
  final int dataVolume;   // MB
  final int validityDays;
  final int price;        // millimes
  final DateTime createdAt;
  final DateTime updatedAt;

  String get formattedPrice => '${(price / 1000).toStringAsFixed(3)} TND';
  String get formattedData => dataVolume >= 1000
      ? '${(dataVolume / 1024).toStringAsFixed(0)} GB'
      : '$dataVolume MB';
  String get formattedValidity => '$validityDays Jours';
}

File: esim-mobile/src/types/offer.ts
export interface Destination {
  id: string;
  country: string;
  countryCode: string;
  coverageType: 'LOCAL' | 'REGIONAL' | 'GLOBAL';
  startingPrice: number;
  offerCount?: number;
  networkType?: string;
  Region: string;
}

export interface Offer {
  id: number;
  country: string;
  countryCode: string;
  coverageType?: 'LOCAL' | 'REGIONAL' | 'GLOBAL';
  dataVolume: string;
  validityDays: number;
  price: number;
  currency: string;
  title: string;
  providerId: string;
  description?: string;
  popularity?: number;
  networkType?: string;
}

export type PlanType = 'data' | 'days';
